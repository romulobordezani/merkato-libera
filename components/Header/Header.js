import style from './Header.style';

const Header = () => {
    return (
        <header style={style}>
            <img src="/static/images/logo-pt__large_plus.png" alt="Mercado Livre"/>
        </header>
    )
};

export default Header;
