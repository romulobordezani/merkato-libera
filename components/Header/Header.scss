.header {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-template-rows: 1fr;
  grid-template-areas: ". contentBox contentBox contentBox contentBox contentBox contentBox contentBox contentBox contentBox .";
}

.SearchBox {
  grid-area: contentBox;
}
