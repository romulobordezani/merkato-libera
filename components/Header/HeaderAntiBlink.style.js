export default {
  background: '#FFE600',
  position: 'sticky',
  width: '100%',
  height: '56px',
  top: 0,
  left: 0,
  zIndex: 1000
};
