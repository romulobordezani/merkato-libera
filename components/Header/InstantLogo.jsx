/**
 * While testing, this img logo was blinking a lot in production, probably because is on Heroku and without a good
 * CDN server, so I converted it to a BASE 64 to get it rendered with HTML on SSR as WORKAROUND :)
 */
const InstantLogo = () => (
  <img
    alt="Mercado Livre"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADEAAAAiCAMAAADBAMEbAAAAA3NCSVQICAjb4U/gAAAA81BMVEX////09fn//wD//ADk5vr/9gDm5+b/7wD/6QD/5wD+4ADU1d3KzvXv1QDExdTfyACsr8ekq+jMtgDArQGtoRySlLSMj56glCuCi9qChauYjAWRhTltdtB/dkdpbp1gab58bxhoZVNdYY5rYzRTV4pHUKtcVidQUWJITYlITYJBRmk5Pnc1PYQ0PI0yOnUwOIksNYAsNX0sNIAqMoQqM4EpMX4mMH4mLn8qL2wmL3wkL3wlL30kLnwkLn4jLn0lLnwkLXwjLXwjLX0iLHsjLHwiK30cJ3kVIYATHW8PHH8IGYkMGXwFEooADZALDk4DDGn///9FxAG9AAAAUXRSTlP/////////////////////////////////////////////////////hP7//xv/6SFQ1hArjjeBW/7HZvS8mbL/1d/m/3NG///9n/7Agf//zgCX/x75AAAACXBIWXMAAAsSAAALEgHS3X78AAAAHHRFWHRTb2Z0d2FyZQBBZG9iZSBGaXJld29ya3MgQ1M26LyyjAAAA05JREFUOI11VG17ojAQDFFAUVREDkRFRJQCWu1plWvhsFKLUmn7/3/NJYBVuLv9AHnZ2cyTzA74KsRiez65D/c4HtzTebsoJoDcbOntjIHIN1kGB9vkxYGxOx/+h1it96MOS0MISZJOA40rbGf0cl79A7E6WTKLkytVHCkiGUDIitZpUUSc7Q4Ns+xqtVRClFgGQjqZkrDSsc85xOpzTCICaTpdKvFjwzRNYywz2aFoV3xfXBEHU9DGcpPGpFBZdjqX6mWiXJcmltrG2ZBm2qp5/3hBHGwKENzQNFS5zasD62NIgCzqfXSSqqojayIRlPOYIhYOlewSXH+oKBIXCngiKN0UV+ckSeLKeEg5swTxWQO3YWJAy9EV3WmBQjQ+MeL1x+0aoWOAkiS3HKUI+fGEEPfE7VLjo1Fu2Xq6RthmAUHdf4GtkF9rabp+ZaPZVOGQJTi1iPwayM27x0a+oAd+GeZQqhf53qQcryfWpaHugge2KY8ts/9fUOPYTf7ccG5NZf4X2CH5QYZXrQkHKMQwz5FAU8pWQLk/t1QeCY3dgV27lEipIhu6pmmKrt8SpxQ0IzTfUNs01mWp/Q7eRZjqDzJOHQhmDd2tcgU1bDyeyJmMofgbeAZJphDGShk1uoLd+L61Gn7MvpqVpa0zWEZtmDWFPBlm9yJoKRa/ec0e9g02bRTIxwfwFVk0nRXgp/PLlWqJ2Lo6+kgfMlNKO6ti75FKNvEga7UqhOOhhI+hKMGmWui5Fa3MzZvfrTiKX7F2n+MeJLP2o+XpsU5gZXSFBlbI3DH4jDUJe/E+UfsyRJBsFR3DWybmpgigNucko5ldDNoZxOEh7cFNGI/YC7NqiWmqE05ydEmyp2M2634askYcbi59vg5iv0NeMOg55bHKtlW1CTO6NKTFIH67u3rJJohio1OB31dAYo/I3AgLomPF0fHu1q+W+yCOrV5qJ2SlcjEi7G90s4fyg5dl3uF+er4fxaHR41lUFGaBnIjvGWEcvfnez798d+U9+2EUR741GvREsSOKvcHIQmWiMHj2rsZ76+2zV9f130IE+44wfPNd93X2b29PQNu1u3d93w+CAH3dF3e9neVTCggMelw+bdae5603T8vH2V/7fwBpSco8lMDxkwAAAABJRU5ErkJggg=="
  />
);

export default InstantLogo;
