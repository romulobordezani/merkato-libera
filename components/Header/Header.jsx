import grid from '../Grid/Grid.scss';
import antiBlinkStyle from './HeaderAntiBlink.style';

import InstantLogo from './InstantLogo';
import SearchBox from '../SearchBox';

const Header = () => {
  return (
    <header className={`${grid.mainGrid}`} style={antiBlinkStyle}>
      <div className={`${grid.contentBox}`}>
        <InstantLogo />
        <SearchBox />
      </div>
    </header>
  );
};

export default Header;
