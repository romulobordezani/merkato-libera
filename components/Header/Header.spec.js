import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

describe('Header', () => {
  it('Snapshot renders default', () => {
    expect(renderer.create(<Header />)).toMatchSnapshot();
  });
});
