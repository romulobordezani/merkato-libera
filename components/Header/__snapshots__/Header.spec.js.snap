// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot renders default 1`] = `
<div
  className="undefined"
  style={
    Object {
      "background": "#FFE600",
      "height": "56px",
      "left": 0,
      "position": "sticky",
      "top": 0,
      "width": "100%",
      "zIndex": 1000,
    }
  }
>
  <div
    className="undefined"
  >
    <div
      className="undefined"
    >
      <div>
        <img
          alt="Mercado Livre"
          onClick={[Function]}
          onMouseEnter={[Function]}
          src="/images/Logo_ML.png"
          srcSet="/images/Logo_ML@2x.png.png 1000w, /images/Logo_ML@2x.png.png  2000w"
        />
      </div>
      <div
        className=" undefined"
        onSubmit={[Function]}
      >
        <form
          autoComplete="on"
        >
          <input
            autoFocus={true}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Nunca dejes de buscar"
            type="text"
          />
          <button
            type="submit"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
